store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java

```diff
@@ -145,7 +146,7 @@ public FederatedStore(@JsonProperty("customPropertiesAuths") final Set<String> c
                           @JsonProperty("storeConfiguredMergeFunctions") final Map<String, BiFunction> storeConfiguredMergeFunctions) {
         Integer i = null;
         while (isNull(i) || ALL_IDS.contains(i)) {
-            i = new Random().nextInt();
+            i = RANDOM.nextInt();
```

Comment:
Unfortunately SpotBugs considers the practice of non field Random to be a violation. If reverted this would need to be suppressed.

So the FederatedStore object is only ever created once for each federated store an instance of Gaffer might use? So normally this constructer is called once per JVM?

Reply:
Yes normally You'd only have one FederatedStore per JVM. I think this should be surpressed and put back.

Reply:
I've put this back to how it was and added the suppression.